Objects/setobject.c

```diff
@@ -1063,36 +1063,67 @@ make_new_set_basetype(PyTypeObject *type, PyObject *iterable)
 /* The empty frozenset is a singleton */
 static PyObject *emptyfrozenset = NULL;
 
+static PyObject *
+make_new_frozenset(PyTypeObject *type, PyObject *iterable)
+{
+    assert(PyType_Check(type));
```

Comment:
I proposed to move the assert in make_new_set, not in make_new_frozenset. So more code will benefit of the assert ;-)

Reply:
Oh, make sense!